lib/utils/files.js

```diff
@@ -38,5 +37,13 @@ files.make_artifacts_folder = function (folder_name) {
   return folder_name;
 };  // Ends files.make_artifacts_folder()
 
+files.below_21 = function (page_id) {
+  return `${ page_id }`.substring(0, 20);
+};
+
+files.below_71 = function (scenario_name) {
+  return `${ scenario_name }`.substring(0, 70);
+};
```

Comment:
Thanks, these explanations help a lot.

There are some conflicting arguments:

> I don't want to decide at each location where these functions are being called how many characters to allow. If we have to update it, we're then stuck updating in multiple places. Also, it lets someone in the future come by and just change it without much thought (which makes me think these need function descriptions too).

This argues towards the existing code, just with different names (TBH `truncate_to_id_max_length` is good here).

> I decided against truncate_to_id_max_length etc. is because I think that when the scenario description isn't in the mix of the path name, maybe we could allow the question id to be longer.

This argues towards deciding the length at each call site; if how long we're truncating things to depends on how we're using it, that should be decided at the call site; anything else is gonna be a lot of extra functions and abstractions for limited benefit.

You've done a lot of great work at making choices and shortening things where you can, but IMO we're going to have to tweak the exact numbers, and I'd lean towards making it easier to do that.

Reply:
Those are fair points. In that case, I think I should get rid of these functions because it's just recreating `.substring()`. A couple less functions to maintain! Yay! Also, I believe `.substring()` doesn't need us to limit the length - it automatically uses whichever is shorter.

Reply:
Adjusted. If you have a mo, I'd appreciate your eye on the changes.